File: src/main/java/com/chidozie/core/gaussianBlur/HorizontalBlurShader.java
package com.chidozie.core.gaussianBlur;

import com.chidozie.core.shaders.ShaderProgram;

public class HorizontalBlurShader extends ShaderProgram {

	private static final String VERTEX_FILE = "gaussianBlur/horizontalBlurVertexShader.vert";
	private static final String FRAGMENT_FILE = "gaussianBlur/blurFragmentShader.frag";
	
	private int location_targetWidth;
	
	protected HorizontalBlurShader() {
		super(VERTEX_FILE, FRAGMENT_FILE);
	}

	protected void loadTargetWidth(float width){
		super.loadFloat(location_targetWidth, width);
	}
	
	@Override
	protected void getAllUniformLocations() {
		location_targetWidth = super.getUniformLocation("targetWidth");
	}

	@Override
	protected void bindAttributes() {
		super.bindAttribute(0, "position");
	}
	
}

File: src/main/kotlin/com/adv/core/gaussianBlur/HorizontalBlurShader.kt
package com.adv.core.gaussianBlur

import com.adv.core.shaders.ShaderProgram

class HorizontalBlurShader : ShaderProgram(VERTEX_FILE, FRAGMENT_FILE) {
    private var location_targetWidth = 0

    fun loadTargetWidth(width: Float) {
        super.loadFloat(location_targetWidth, width)
    }

    override fun getAllUniformLocations() {
        location_targetWidth = super.getUniformLocation("targetWidth")
    }

    override fun bindAttributes() {
        super.bindAttribute(0, "position")
    }

    companion object {
        private const val VERTEX_FILE = "gaussianBlur/horizontalBlurVertexShader.vert"
        private const val FRAGMENT_FILE = "gaussianBlur/blurFragmentShader.frag"
    }
}
